build.gradle.kts

```diff
@@ -0,0 +1,191 @@
+import com.hierynomus.gradle.license.LicenseBasePlugin
+import de.marcphilipp.gradle.nexus.NexusPublishPlugin
+import net.ltgt.gradle.errorprone.ErrorPronePlugin
+import net.ltgt.gradle.errorprone.errorprone
+
+buildscript {
+    repositories {
+        mavenCentral()
+        maven("https://oss.sonatype.org/content/repositories/snapshots/")
+        jcenter()
+    }
+    dependencies {
+        classpath("com.bmuschko", "gradle-nexus-plugin", "2.3.1")
+    }
+}
+
+plugins {
+    `java-library`
+    signing
+    id("checkstyle")
+    id("com.github.hierynomus.license") version "0.15.0"
+    id("com.github.johnrengelman.shadow") version "6.1.0"
+    id("de.marcphilipp.nexus-publish") version "0.4.0"
+    id("net.ltgt.errorprone") version "1.3.0"
+}
+
+checkstyle {
+    configFile = file("config/checkstyle/checkstyle.xml")
+}
+
+buildGroups("Minecraft", "Discord", "IRC")
+
+gradle.taskGraph.whenReady {
+    gradle.taskGraph.allTasks.forEach {
+        if (it.project.name.contains("example")) {
```

Comment:
Should we create a group for the examples as well, to be consistent?

Reply:
the example are a bit different as we *only* want to build them when we include the flag, as opposed to the groups where normally you build everything but the group allows building a smaller subsection.

Reply:
fair